### PR TITLE
feat: add diagnostics source name

### PR DESCRIPTION
**Pull request by kylo252**
# Description

- Add diagnostics' source name and to help distinguish where it's coming from.
- Add diagonstics' code to make it easier for the user to turn it off if the provider supports that.

## Fixes

`setup_handlers()` from lsp wasn't getting called also.

## How Has This Been Tested?

image

### CODE REVIEW

File: lua/lsp/init.lua [hunk 1]
```diff
@@ -208,6 +231,7 @@ function lsp_config.common_on_attach(client, bufnr)
     lvim.lsp.on_attach_callback(client, bufnr)
   end
   lsp_highlight_document(client)
+  lsp_config.setup_handlers()
```
Comment (Tastyep) [inline lua/lsp/init.lua:234]: `setup_handlers` is already called from init.lua

File: lua/lsp/init.lua [hunk 2]
```diff
@@ -65,11 +65,34 @@ end
...
+    local bufnr = vim.uri_to_bufnr(uri)
+
+    if not bufnr then
+      return
+    end
+
+    local diagnostics = params.diagnostics
+
+    for i, v in ipairs(diagnostics) do
+      diagnostics[i].message = string.format("%s: %s", v.source, v.message)
```
Comment (Tastyep) [inline lua/lsp/init.lua:85]: Have you considered adding the code too ? Maybe smth like `"%s [%s]: %s"`
See [here](https://github.com/jose-elias-alvarez/null-ls.nvim/pull/57/files#diff-c2c076648e1b1e04ff69ac1030444a8ffbfda480d12fe8771b719a77cba92d7fR142) for examples of code values.

**Review by Tastyep — CHANGES_REQUESTED** [review]
Looks nice :+1:

**Comment by ChristianChiarulli** [comment]
there is a dedicated file for handlers now, could you use that, this looks like a really useful feature.

**Comment by kylo252** [comment]
> there is a dedicated file for handlers now, could you use that, this looks like a really useful feature.

I'll fix this once #1078 is merged.

**Comment by ChristianChiarulli** [comment]
May need a slight refactor now, but looks good.

File: lua/lsp/init.lua (hunk 2 above)
Comment (kylo252) [reply lua/lsp/init.lua:85]: > Have you considered adding the code too ? Maybe smth like `"%s [%s]: %s"`
> See [here](https://github.com/jose-elias-alvarez/null-ls.nvim/pull/57/files#diff-c2c076648e1b1e04ff69ac1030444a8ffbfda480d12fe8771b719a77cba92d7fR142) for examples of code values.

I'll try to see what I can come up with.
Thanks for all the amazing work for null-ls btw :)

**Review by Tastyep — APPROVED** [review]
Looks great :+1:

**Pull request merged**